diagnostic_feedback/public/js/src/student/student_quiz.js

```diff
@@ -144,15 +145,42 @@ function StudentQuiz(runtime, element, initData) {
 
           if (success && response.student_result) {
             //log event for quiz finish
-            event_data.event_type = 'xblock.diagnostic_feedback.quiz.finish';
+
+            var event_data = {
+              event_type: 'xblock.diagnostic_feedback.quiz.finish',
```

Comment:
@jareerahsan Should this event be fired before the `xblock.diagnostic_feedback.quiz.question.submitted` event? I'm not sure but it seems like submitting a question needs to happen before a student can finish a quiz (and the order in which events are fired should reflect that).

Even if that's not the case, you could simplify the structure of the `success` callback a bit:

```js
if (success) {
  // Logic for successful submit
  if (!isLast) {
    // Logic for loading next question
  } else if (response.student_result) {
    // Logic for finished quiz
  }
} else {
  // Logic for submit error
}